lib/datatables_crud.rb

```diff
@@ -6,32 +6,22 @@ def current_path
   File.dirname(__FILE__)
 end
 
-def include_file(filename)
-  filename = filename.sub(current_path, '')
-  require File.join(current_path, File.dirname(filename), File.basename(filename, '.rb'))
-end
-
-autoload_dirs = %w{datatables helpers support}
-
-autoload_dirs.each do |dir|
-  Dir.glob(File.join(current_path, 'datatables_crud', dir, '**/*.rb')).sort.each do |file|
-    include_file file
-  end
-end
-
 # load localizations
 Dir.glob(File.join(current_path, 'datatables_crud', 'config', 'locales', '**/*.yml')).each do |locale_file|
   I18n.load_path << locale_file
 end
 
 module DatatablesCRUD
+  require "#{__dir__}/datatables_crud/datatables/abstract_datatable"
+  require "#{__dir__}/datatables_crud/support/datatables_crud/controller"
 
   class Railtie < ::Rails::Railtie
-    initializer 'datatables_crud.init' do |app|
-      Dir.glob(File.join(current_path, 'datatables_crud', 'controllers', '**/*.rb')).sort.each do |file|
-        include_file file
-      end
+    configure do
+      # XXX: Is it coming too late? Zeitwerk does not seem to take care about these entries when the app is already running.
+      # Or the names / nesting are wrong?
+      ActiveSupport::Dependencies.autoload_paths << "#{__dir__}/datatables_crud/controllers"
+      ActiveSupport::Dependencies.autoload_paths << "#{__dir__}/datatables_crud/datatables"
+      # ActiveSupport::Dependencies.autoload_paths << "#{__dir__}/datatables_crud/support"
```

Comment:
Does "/support" have to be commented out?

Reply:
Good point! I suspect that no, it does not. I have commented that out, because there is only one file in there, and it had been explicitly `require`d above.

As I'm thinking now, the names are probably indeed wrong - the `CRUD` is (probably) not declared as an acronym in the I18n, and that's why Zeitwerk does not load it.

I think I would like to test it one more time by the next opportunity. :thinking:

Reply:
https://miceportal.atlassian.net/browse/MP-47616 - I've created a ticket, so that whoever would merge (it could even be me), would also spend a short while looking again at why that autoloading was necessary.